src/bcf_clean_overpar.cpp

```diff
@@ -339,6 +339,10 @@ List bcfoverparRcppClean(NumericVector y_, NumericVector z_, NumericVector w_,
   size_t save_ctr = 0;
   bool verbose_itr = false; 
 
+  // In BCF 1.3 on CRAN, this is done inside the MCMC Loop
+  // As it is done in this commit https://github.com/jaredsmurray/bcf/commit/c7bb4cbb851b1004e0ad759c1360d43b5cfe1729
+  // I believe that constance added this here before the merge in this commit
+  // https://github.com/mathematica-mpr/bcf-1/commit/9108cde550be83d9cf5bef2ac89e37f2241b5068#diff-6d2dc2eeed05be5f999493994f51f44cf947ecda0b640135b1565262ed8181cf
   if(prior_sample) {
     for(int k=0; k<n; k++) y[k] = gen.normal(allfit[k], sigma);
   }
```

Comment:
I don't have an intuition about what's supposed to happen when "prior_sample" is true, but in the github version this is outside the MCMC and in the CRAN version it's inside.  @e-lipman, @MPRdthal, do you have any guess what this is doing and if it should be doing it inside or outside the MCMC loop?

Reply:
I dunno exactly what's going on, but my vauge sense:
Before the MCMC block, allfit (which is what I think of as yhat) is intialized and filled in - maybe just based on the priors and not the data?
if prior_sample is true, we replace our actual data with random data drawn based on allfit, and we'll be fitting the model to the random data. That sounds like a standard stan-ish "sample data from the prior" hack, kinda? I'm not really sure though.

So then if that's outside the loop, we're drawing the data once, and then fitting the model to that one random draw. If it's inside the loop we're replacing the data each time, and each iteration fits new random data. I have no idea what intended behavior is, but outside the loop makes more sense to me since you're only doing it once? But maybe you'd want to redraw each time.